source/extensions/retry/host/other_hosts/config.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "envoy/upstream/retry.h"
+
+#include "extensions/retry/host/other_hosts/other_hosts.h"
+#include "extensions/retry/host/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+
+class OtherHostsRetryPredicateFactory : public Upstream::RetryHostPredicateFactory {
+public:
+  void createHostPredicate(Upstream::RetryHostPredicateFactoryCallbacks& callbacks,
+                           const ProtobufWkt::Struct&) override {
```

Comment:
@htuch I think I was too late to point out this in #4385, we should avoid leaking `Struct` to extension interfaces? This is similar to #3155, and given we are already facing some config performance issue so it makes sense.

Reply:
Let's continue this discussion in https://github.com/envoyproxy/envoy/issues/4475. We do want to move in this direction, but we want to maintain a consistent and unsurprising API surface for extensions.

Reply:
@snowp is this the right PR to followup on #4475 tactical changes or should we review as is and you'll do that in a followup?

Reply:
I think it makes sense to do in a follow up, that way I can do both extensions at the same time without cluttering this PR

Reply:
Actually, considering #4385 has still not been reviewed (and this introduces the other Struct usage) I think I'll just do it in each PR separately

Reply:
Turns out it's more elaborate than I thought - I'll update this PR to use `MessagePtr` instead of `Struct` and then follow up with another PR setting up a base factory to convert that into a config proto. This PR isn't relying on parsing a config proto, so I think it can be merged without having all that set up

Sorry for being so indecisive!